### PR TITLE
feat(upgrade): add command to delete upgrade-resources

### PR DESCRIPTION
**Pull request by sinhaashish**

### CODE REVIEW

**Comment by sinhaashish** [comment]
bors try

File: k8s/upgrade/src/upgrade_resources/upgrade.rs [hunk 1]
```diff
@@ -585,28 +589,34 @@ pub async fn get_release_name(ns: &str) -> Result<String, Error> {
...
+                            .succeeded
+                            .as_ref()
+                            .ok_or_else(|| anyhow::anyhow!("upgrade job has not completed yet."))?,
+                        1
+                    ) {
+                        return Ok(true);
+                    }
+                }
+                None => {
+                    println!("Upgrade job {job_name} in name space {ns} does not exist");
```
Comment (niladrih) [inline k8s/upgrade/src/upgrade_resources/upgrade.rs:613]:
```suggestion
                    eprintln!("Upgrade job {job_name} in name space {ns} does not exist");
```

File: k8s/upgrade/src/upgrade_resources/upgrade.rs [hunk 2]
```diff
@@ -27,11 +27,28 @@ use kube::{
...
+    pub async fn delete(ns: &str) {
+        // Delete upgrade resources once job completes
+        match upgrade_job_completed(ns).await {
+            Ok(job_completed) => {
+                if job_completed {
+                    UpgradeResources::delete_upgrade_resources(ns).await;
+                }
+            }
+            Err(error) => {
+                println!("Failure: {error}");
```
Comment (niladrih) [inline k8s/upgrade/src/upgrade_resources/upgrade.rs:48]:
```suggestion
                eprintln!("Failure: {error}");
```

File: k8s/upgrade/src/upgrade_resources/upgrade.rs [hunk 3]
```diff
@@ -27,11 +27,28 @@ use kube::{
...
+
+impl Upgrade {
+    /// Delete the upgrade resources
+    pub async fn delete(ns: &str) {
+        // Delete upgrade resources once job completes
+        match upgrade_job_completed(ns).await {
+            Ok(job_completed) => {
+                if job_completed {
+                    UpgradeResources::delete_upgrade_resources(ns).await;
+                }
```
Comment (niladrih) [inline k8s/upgrade/src/upgrade_resources/upgrade.rs:45]: Maybe an else scenario for when the job is not complete is not complete is required. I feel this lacks feedback saying that the running job cannot be deleted.

Maybe also a `--force` or `-f` flag to forcibly delete.

**Review by niladrih — COMMENTED** [review]
Left few comments..

File: k8s/upgrade/src/upgrade_resources/upgrade.rs (hunk 3 above)
Comment (sinhaashish) [reply k8s/upgrade/src/upgrade_resources/upgrade.rs:45]: Force can be added in another PR.

**Review by niladrih — APPROVED** [review]
LGTM

File: k8s/plugin/src/resources/mod.rs [hunk 4]
```diff
@@ -34,4 +34,7 @@ pub enum Operations {
     Dump(DumpArgs),
     /// `Upgrade` the deployment.
     Upgrade(UpgradeArgs),
+    /// `Delete` the upgrade resources.
```
Comment (tiagolobocastro) [inline k8s/plugin/src/resources/mod.rs:37]:
```suggestion
    /// `Delete` the resources.
```

File: k8s/plugin/src/resources/mod.rs [hunk 5]
```diff
@@ -34,4 +34,7 @@ pub enum Operations {
...
+    #[clap(subcommand)]
+    Delete(Upgrade),
```
Comment (tiagolobocastro) [inline k8s/plugin/src/resources/mod.rs:39]: Please create a new `DeleteResources` enum which contains the Upgrade only for now, and add that type here to allow for future expansion.

**Review by tiagolobocastro — APPROVED** [review]

**Comment by sinhaashish** [comment]
bors merge

**Pull request merged**